Fill.py

```diff
@@ -136,33 +136,93 @@ def fill_restrictive(world: MultiWorld, base_state: CollectionState, locations:
     itempool.extend(unplaced_items)
 
 
+def fast_fill(world: MultiWorld, locations: typing.List[Location],
+                     itempool: typing.List[Item], single_player_placement: bool = False) -> None:
+    unplaced_items: typing.List[Item] = []
+    placements: typing.List[Location] = []
+    swap = 0
+    swapped_items: typing.Counter[typing.Tuple[int, str]] = Counter()
+    while locations and itempool:
```

Comment:
the fill then swap code looks very similar to the other one; maybe that can be turned into a common function?

Reply:
I'm not seeing what could easily be done to that end, every section has differences